### DISCUSSION TITLE
Running on M1 Mac

### DISCUSSION THREAD

**Discussion by conradg** [marcoppasini/musika#8] [General]
To get this to work on macOS with Metal and an M1 I had to use the following requirements in `requirements.txt`

```
gradio==3.3.1
librosa==0.8.1
matplotlib==3.4.3
numpy==1.23.4
scipy==1.9.3
tensorboard==2.10.0
tensorflow-macos==2.10.0
tensorflow-metal==0.6.0
tqdm==4.62.3
pydub==0.25.1
```

key change is adding the[ tensorflow plugin for macos](https://developer.apple.com/metal/tensorflow-plugin/), as using regular pip `tensorflow` package was giving the following error
```
ERROR: Could not find a version that satisfies the requirement tensorflow (from versions: none)
ERROR: No matching distribution found for tensorflow
```

this fixes the install :)

## performance

23 seconds of music can be generated in about 2 seconds

<img width="1537" alt="image" src="https://user-images.githubusercontent.com/4610193/197214700-fdfc8564-f6ea-4015-abc1-711a5a06c1fa.png">

**Comment by conradg** [comment]
```python musika_train.py --train_path ../encodings --load_path checkpoints/misc --lr 0.00004 --mixed_precision False --xla False``` used this command to train - replace encodings with your encoded samples

**Comment by jbremz** [comment]
Thanks! This was really helpful 🙏 

Also seeing similar speeds 🙂 

Just to add: I had to update from macOS 12.0 (I think) to 12.6 because tensorflow-macos was still having some problems

**Comment by thelatebphelium** [comment]
thanks @conradg !!  i was going round in circles... :)

**Comment by AI-Guru** [comment]
I am running into similar problems.

Getting this error message. tensorflow-macos and tensorflow-metal have the same version as mentioned above.

```
T.train(ds, models_ls)
  File "/Users/tristan/Development/musika-experiments/musika/train.py", line 195, in train
    dloss_tr, dloss_tf, dloss_id, gloss_t = train_tot(a, ema, models_ls=models_ls)
  File "/Users/tristan/miniforge3/lib/python3.10/site-packages/tensorflow/python/util/traceback_utils.py", line 153, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Users/tristan/miniforge3/lib/python3.10/site-packages/tensorflow/python/eager/execute.py", line 54, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.UnimplementedError: Could not find compiler for platform METAL: NOT_FOUND: could not find registered compiler for platform METAL -- was support for that platform linked in? [Op:__inference_train_tot_22153]
```
